components/workflow-mgt/org.wso2.carbon.identity.workflow.mgt/src/main/java/org/wso2/carbon/identity/workflow/mgt/WorkflowService.java

```diff
@@ -369,4 +377,62 @@ public boolean eventEngagedWithWorkflows(String eventType) throws InternalWorkfl
         }
 
     }
+
+    /**
+     * Returns array of requests initiated by a user.
+     *
+     * @param user
+     * @return
+     * @throws WorkflowException
+     */
+    public WorkflowRequestDTO[] getRequestsCreatedByUser(String user) throws WorkflowException {
+
+        return workflowRequestDAO.getRequestsOfUser(user);
+    }
+
+    /**
+     * Update state of a existing workflow request
+     *
+     * @param requestId
+     * @param newState
+     * @throws WorkflowException
+     */
+    public void updateStatusOfRequest(String requestId, String newState) throws WorkflowException {
+        if (WorkflowRequestStatus.DELETED.toString().equals(newState)) {
+            workflowRequestDAO.updateStatusOfRequest(requestId, newState);
+        }
+        requestEntityRelationshipDAO.deleteRelationshipsOfRequest(requestId);
+    }
+
+    public WorkflowRequestDTO[] getRequestsFromFilter(String user, String beginDate, String endDate, String
+            dateCategory) throws WorkflowException {
+
+        SimpleDateFormat dateFormat = new SimpleDateFormat("MM/dd/yyyy");
+        Timestamp beginTime;
+        Timestamp endTime;
+
+        try {
+            Date parsedBeginDate = dateFormat.parse(beginDate);
+            beginTime = new java.sql.Timestamp(parsedBeginDate.getTime());
+        } catch (ParseException e) {
+            long millis = 0;
+            Date parsedBeginDate = new Date(millis);
+            beginTime = new java.sql.Timestamp(parsedBeginDate.getTime());
+        }
+        try {
+            Date parsedEndDate = dateFormat.parse(endDate);
+            endTime = new java.sql.Timestamp(parsedEndDate.getTime());
+        } catch (ParseException e) {
+            Date parsedEndDate = new Date();
+            endTime = new java.sql.Timestamp(parsedEndDate.getTime());
+        }
+        log.error(endTime);
```

Comment:
Thanks, this is already fixed with https://github.com/wso2/carbon-identity/pull/834